---
title: Calculate sample size or power for published microbiome studies with IMPACTT
  guidelines
author: "David Xin Zhao"
date: "Last edited `r format(Sys.time(), '%d %B %Y')`"
knit: (function(inputFile, encoding) {
      out_dir <- 'docs';
      rmarkdown::render(inputFile,
                        encoding=encoding,
                        output_file=file.path(dirname(inputFile), out_dir, 'index.html'))})
output:
  html_document:
    # theme: cosmo
    highlight: pygments
    df_print: paged
    toc: TRUE
    toc_float: TRUE
    collapsed: FALSE
    number_sections: TRUE
    fig_width: 7
    fig_height: 6
    fig_caption: TRUE
editor_options: 
  markdown: 
    wrap: 72
bibliography: references.bib
---

<html>

<head>

```{=html}
<style>

h1{
 color: #055C9D;
 font-family: Georgia;
 font-size: 200%
}


h2{
 color: #055C9D;
 font-family: helvetica;
 font-size: 150%
}

h3{
 color: #055C9D;  
 font-family: helvetica;
 font-size: 120%; 
}

p {
 color: #333333;
 font-family: helvetica;
 font-size: 100%;
}

.blackbox {
  padding: 1em;
  background: green;
  color: black;
  border: 2px solid orange;
  border-radius: 10px;
}

.center {
  text-align: center;
}

</style>
```
</head>

</html>

```{r setup, include = FALSE}
# set options for the entire document 
knitr::opts_chunk$set(fig.align = 'center', 
                      fig.height=6, fig.width=8,
                      dev="png",
                      echo=TRUE, #display code in output document 
                      error=FALSE,
                      collapse = FALSE, 
                      message=FALSE) #stop render when error occurs   
```

```{r}

library(tidyverse)





```

## Problem

Calculating the sample size appropriately is critical in planning a
valid microbiome study. Insufficient sample size will cause poor power
in a hypothesis testing while more-than-enough sample size will cause
resource wasting or even animal/ human ethics problem.

Power is the probability to correctly reject a null hypothesis when it
is false, which indicates how credible a microbiome study is.

Microbiome data has special characteristics different from common data
set, which thus require special methods to calculate sample size or
power.

## Solution

IMPACTT experts provided detailed guidelines in 2021-2022
[@ferdous2022][@beta-div2022], helping microbiome researchers calculate
sample size and power appropriately.

## Project outline

Steps to calculate sample size or power in published microbiome studies:

1.  Identify the research question/ statistical hypothesis
2.  Observe dependent variable distribution and grouping variables
3.  Consider the simple hypothesis test
4.  Determine the standard formulation
5.  Extract parameters available from paper
6.  Implement the formulation

Application of the IMPACTT guideline to published work:

1.  Beta-diversity: smoker study[@chen2012]; C-section study[@madan2016]
2.  Alpha-diversity: Canadian infants[@azad2013] and IPA[@chen2021]
3.  Differential abundance analysis of a taxon: vitamin D
    study[@zhao2023]
4.  Microbial cluster membership: food sensitization[@tun2021]
5.  Taxon membership (colonization with a taxon): C. difficile
    colonization[@drall2020]; B. longum subsp. infantis
    colonization[@chen2021]
6.  Longitudinal study: long-term diet and enterotypes[@wu2011]

## R codes

In this project, I applied the effective framework to real dataset,
while practicing R coding skills and familiarizing statistical concepts.

### Worked example: Distance matrix (beta-diversity)

Commonly used measures of microbiota beta-diversity include:

-   Unweighted UniFrac (phylogeny-aware, emphasis on rare taxa)

-   Weighted UniFrac (phylogeny-aware, emphasis on highly abundant taxa)

-   Generalized UniFrac (UniFrac variant, powerful in detecting changes
    of moderately abundant taxa)

-   Jaccard

-   Bray-Curtis

Scenario 1. Mean and standard deviations of distance is available in the
published paper; assume distance is well-behaved (i.e., normally
distributed distance for pairs in the same group)

Calculated sample size for the beta-diversity comparison in the
published study in 2016, Effects of Cesarean delivery and formula
supplementation on the intestinal microbiome of six-week old infants.

Research question: Is the microbiota structure in vaginal-delivery
infants different that in C-sectional infants?

Statistical hypothesis: distance within group is equal to that between
group

Mean and standard deviations is available in the published paper. Assume
UniFrac beta diversity is normal distributed, so that Formula A (in the
guideline article) is suitable to estimate the required sample size or
power.

```{r extract-means-and-std}

extract_unifrac <- data.frame(variable = c("n", "mean", "std"),
                              vaginal = c(70, 0.5613, 0.0026),
                              c_section = c(32, 0.5587, 0.0046)) 

extract_unifrac  # print the table  

```

```{r beta-distance}

# calculate delta based on parameters in the smaller group 
delta <- (extract_unifrac[2,2] - extract_unifrac[2,3])/extract_unifrac[3,3] 

# assume alpha = 5%, 1-beta = 80%, then required sample size for the C-section group:
n2 <- ceiling(2*(qnorm(0.025, lower.tail = F) + qnorm(0.2, lower.tail = F))^2/delta^2) 

# the required sample size for the vaginal delivery group: 
n1 <- ceiling(70/32*n2)  


```

Scenario 2. Pseudo F or R-square is available in the published paper
(means and standard deviations of distance measures not available), then
use Equation G (in the guideline article) to calculate required sample
size based on correlations.

The published paper[@forbes2018], Association of exposure to formula in
the hospital and subsequent infant feeding practices with gut microbiota
and risk of overweight in the first year of life, provided pseudo F
values when comparing UniFrac distance measures.

Pseudo F = 8.11 for unweighted UniFrac, while pseudo F = 8.36 for
weighted UniFrac (compare No BF vs Partial BF). No BF group has 225
infants while partial BF group has 367 infants.

```{r Equ-G}

# for unweighted UniFrac distance measure

f <- 8.11/(225 + 367 -2)

r_square <- f^2/(1+f^2)  # effect size 

rho <- sqrt(r_square)  

n <- 3 + (1.96 + 0.84)^2/(log((1+rho)/(1-rho))/2)^2 

n

```

Distribution assumption: nonparametric distribution

Scenario 3: distance matrix is known

Scenario 4: distance matrix is unknown - simulate distance matrix (Kelly
et al.)

distribution assumption: Dirichlet-Multinomial distribution (La Rosa et
al.) using HMP R package

### Case study: Alpha-diversity

Alpha diversity refers to the number of species present in an ecosystem
(richness) as well as the frequency of occurrence of each type of
organisms (evenness).

Commonly used metrics include:

-   Shannon

-   Inverse Simpson

-   Simpson

-   Chao indices

-   Faith's phylogenetic diversity (Faith's PD)

```{r alpha-diversity}


## distribution assumption: normal distribution 

# method 1: Equation A  (explanatory variable is binary) 

# method 2: Equation G for correlation (explanatory variable is continuous)


# convert medians and interquartile ranges to means and standard deviations  

```

### Case study: Differential analysis of taxon abundance

Two types of hypothesis tests:

-   Compare mean abundance of a taxon

-   Compare subject proportion with abundance over a chosen threshold

Type I error threshold, alpha should not need adjustment for multiple
testing, if testing on only single taxon.

Type I error threshold, alpha should be adjusted for multiple testing,
if a study plans to test association at all taxa.

alpha\^star = alpha / M (where M is the planned number of tests)

```{r taxon abundance}

## distribution assumption: ?

# Equation D and E in Table 2 



## distribution assumption: non-normal distribution 

# convert median and IQR to mean and standard deviation (suitable for phylum or higher classification levels) 


```

### Case study: Testing higher or lower rates of cluster membership between groups

```{r cluster membership}

# Equation C an E in table 2 


```

### Case study: Testing higher or lower rates of taxon membership between groups (i.e. colonization with a microbe)

To determine sample sizes for colonization with specific microbiota
(presence/ absence or yes/ no), one can use the equations for
proportions or odds ratios.

```{r taxa colonization}

# Equation C in table 2 



```

### Case study: Longitudinal study (SIMR: An R package for power analysis of generalized linear mixed models by simulation)
